### PR TITLE
Store: Add user facing fields for Google Analytics for eCommerce

### PR DESCRIPTION
**Pull request by allendav**
...to site settings when WooCommerce is active on the site

See #17105 

See also https://github.com/Automattic/jetpack/pull/7817

To test:
```
* You'll need three sites - a WPCOM site with a Business plan, a site with Jetpack 5.3 and a Professional plan and a site with Jetpack 5.4-beta3 or newer and a Professional plan

* First, start with the WPCOM site with a Business Plan (but not an Atomic site)
* Navigate to http://calypso.localhost:3000/settings/traffic/DOMAIN
* Scroll down to Google Analytics
* Make sure you can edit the tracking ID (the UA... number)
* Make sure if you View Source for that site, you see the updated tracking ID

* Repeat for the Jetpack site with the running Jetpack 5.3 with a Professional Plan
* Navigate to http://calypso.localhost:3000/settings/traffic/DOMAIN
* Scroll down to Google Analytics
* Make sure you can edit the tracking ID (the UA... number)
* Make sure if you View Source for that site, you see the updated tracking ID

* Lastly, repeat for the Jetpack site with the running Jetpack 5.4-beta3 or newer with a Professional Plan
* Start with WooCommerce deactivated
* Navigate to http://calypso.localhost:3000/settings/traffic/DOMAIN
* Scroll down to Google Analytics
* Make sure you can edit the tracking ID (the UA... number)
* Make sure if you View Source for that site, you see the updated tracking ID
* Make sure you can toggle Anonymize IP on/off
* Make sure if you View Source for that site, you see the anonymize flag when enabled
* Make sure you DO NOT see the Basic Store Analytics settings

* Activate WooCommerce on that site
* Navigate to http://calypso.localhost:3000/settings/traffic/DOMAIN
* Scroll down to Google Analytics
* Make sure you DO see the Basic Store Analytics settings
* Enable tracking of Purchase Transactions
* In another tab, navigate to your Google Analytics Dashboard
* In your Google Analytics Dashboard, go to Admin for your site (it looks like a gear in the lower left corner)
* Under View select Ecommerce Settings
* Enable Ecommerce if you haven't already. DO NOT ENABLE ENHANCED ECOMMERCE AT THIS TIME.
* Open an incognito window to your site and complete a purchase
* Toggle back to the Google Analytics dashboard and go to Conversions and then Ecommerce
* Set the date range to today if needed
* Make sure you see the sale and that the details are correct

* Go back to the tab with http://calypso.localhost:3000/settings/traffic/DOMAIN
* Enable tracking Add to Cart Events
* On the incognito window, visit a few product pages and add some products to your cart.
* Go back to the tab with your Google Analytics Dashboard
* In your Google Analytics Dashboard, go to Real-time and then Events
* Make sure you see the addition of the products and that the details are correct
```

### CODE REVIEW

File: client/my-sites/site-settings/form-analytics-stores.jsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+			</div>
+		);
+	}
+
+	renderBasicSettings = ( disableAll ) => {
+		const { translate } = this.props;
+		const settings = [
+			{
+				key: 'track_purchases',
+				label: translate( 'Purchase transactions' )
```
Comment (a8ci18n) [inline client/my-sites/site-settings/form-analytics-stores.jsx]: Hi! I've found a possible matching string that has already been translated **37** times:
`translate( 'Transactions' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by allendav** [comment]
cc @kellychoffman / @jameskoster for style tweaks / clean up

File: client/my-sites/site-settings/form-analytics-stores.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	}
+
+	renderSettings = ( settings, disableAll ) => {
+		const { fields } = this.props;
+
+		return (
+			<div>
+				{
+					settings.map( setting =>
+						<CompactFormToggle
```
Comment (justinshreve) [inline client/my-sites/site-settings/form-analytics-stores.jsx]: React gripes about this not having a `key` prop.

> Warning: Each child in an array or iterator should have a unique "key" prop. Check the render method of `FormAnalyticsStores`

File: client/my-sites/site-settings/form-analytics.jsx [hunk 3]
```diff
@@ -168,6 +210,34 @@ class GoogleAnalyticsForm extends Component {
 									{ translate( 'Where can I find my Tracking ID?' ) }
 								</ExternalLink>
 							</fieldset>
+							{ showAnonymizeIP &&
+								<fieldset>
+									<CompactFormToggle
+										checked={ fields.wga ? fields.wga.anonymize_ip : false }
```
Comment (justinshreve) [inline client/my-sites/site-settings/form-analytics.jsx]: We should make sure to cast this as a bool before passing. Enabling it shows the correct status, but I get the following console warning:

> Warning: Failed prop type: Invalid prop `checked` of type `string` supplied to `FormToggle`, expected `boolean`.

**Comment by jameskoster** [comment]
Would you mind including screenshots of the UI so we an see if any design tweaks are needed? Looks like we're just adding some inputs here so probably not?

**Comment by allendav** [comment]
@jameskoster 

Jetpack 5.3:
jetpack-5-3

Jetpack 5.4 with WooCommerce:
jetpack-5-4-with-wc

Jetpack 5.4 without WooCommerce:
jetpack-5-4-no-wc

**Comment by allendav** [comment]
@justinshreve wrote

> The traffic settings page in general returns a huge amount of warnings/notices separate from this PR (I checked on master). I did find two minor warnings related to this PR though. I've highlighted where they are occurring.

Yeah, I noticed those too - I've written that up as https://github.com/Automattic/wp-calypso/issues/18592

File: client/my-sites/site-settings/form-analytics-stores.jsx (hunk 2 above)
Comment (allendav) [reply client/my-sites/site-settings/form-analytics-stores.jsx]: Fixed in https://github.com/Automattic/wp-calypso/pull/17985/commits/51f29ed52a49f45dc7fd8b45a560a5e05acc8a39

File: client/my-sites/site-settings/form-analytics.jsx (hunk 3 above)
Comment (allendav) [reply client/my-sites/site-settings/form-analytics.jsx]: Fixed in https://github.com/Automattic/wp-calypso/pull/17985/commits/51f29ed52a49f45dc7fd8b45a560a5e05acc8a39

**Comment by allendav** [comment]
@justinshreve - ok - all rebased and pretty - ready for another review

**Comment by kellychoffman** [comment]
Screenshots look good! Only one request for `Jetpack 5.4 with WooCommerce`. Can we have the same top padding above the 'Google Analytics is a free service..' paragraph as the seen on the rest of them? Otherwise, is 👍 .

**Comment by justinshreve** [comment]
@allendav Recent changes are looking good to me. After the padding update, let's 🚢

**Comment by jameskoster** [comment]
Thanks @allendav.

The alignment of the toggles/labels looks off, but that's separate issue territory. Other than that I think we just need to add some spacing beneath the "Add to cart events" option.

**Comment by allendav** [comment]
@justinshreve @jameskoster - thanks for your feedback. I added padding in https://github.com/Automattic/wp-calypso/pull/17985/commits/e4837cfb39a34d1a8f12f708039ce5b2d8475f06

Will commit as soon as circleci is done/clear

**Pull request merged**